lib/ecto/type.ex

```diff
@@ -547,6 +547,10 @@ defmodule Ecto.Type do
 
   """
   @spec cast(t, term, map) :: {:ok, term} | :error
+  def cast({:embed, embed}, value, id_types) do
+    cast_embed(Ecto.Embedded.normalize_changeset(embed, nil), value, id_types)
```

Comment:
As mentioned above, let's get rid of normalize_changeset. Just assume the :changeset will be an atom.